tests/unit/configuration_test.py

```diff
@@ -1888,6 +1907,86 @@ def test_parse_compression_algorithm_invalid_compression_level(self):
             msg = "Expected %s for algorithm %s" % (valid_level_max, compression_type)
             self.assertEqual(config.compression_level, valid_level_max, msg)
 
+    @skipIf(
+        sys.version_info < (2, 7, 0),
+        "Skipping under Python < 2.7 (zstandard not available for Python 2.6)",
+    )
+    @mock.patch("scalyr_agent.configuration.zstandard", mock.Mock())
+    def test_zstandard_is_used_by_default_if_python_library_is_available(self):
+        # zstandard Python library is available, should use that
+        self._write_file_with_separator_conversion(
+            """{
+                api_key: "hi there",
+            }
+            """
+        )
+
+        config = self._create_test_configuration_instance()
+        config.parse()
+
+        self.assertEqual(config.compression_type, "zstandard")
+
+    @skipIf(
+        sys.version_info < (2, 7, 0),
+        "Skipping under Python < 2.7 (zstandard not available for Python 2.6)",
+    )
+    @mock.patch("scalyr_agent.util.zstandard", None)
+    @mock.patch("scalyr_agent.configuration.DEFAULT_COMPRESSION_ALGORITHM", "deflate")
+    @pytest.mark.forked
+    @pytest.mark.relies_on_reload
+    @pytest.mark.needs_to_run_isolated
```

Comment:
If we have a mark for `relies_on_reload`, we should always run it in isolation, correct?  Is there a way to enforce that by updating the definition for  `relies_on_reload` to make sure they are run in isolation?  I don't think `forked` implies `needs_to_run_isolated` since you have test cases down below that fork but don't specify to run in isolation.

I'm also wondering if there's a good way to make sure folks know to use these annotations.  It seems adding documentation to `reload` should be one way.  But, for `forked`, I don't know if there's one clear place we can document this.